cmd/kubeadm/app/cmd/phases/controlplane.go

```diff
@@ -0,0 +1,159 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package phases
+
+import (
+	"path/filepath"
+
+	"github.com/spf13/cobra"
+
+	kubeadmapi "k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm"
+	kubeadmapiext "k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm/v1alpha1"
+	"k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm/validation"
+	kubeadmconstants "k8s.io/kubernetes/cmd/kubeadm/app/constants"
+	controlplanephase "k8s.io/kubernetes/cmd/kubeadm/app/phases/controlplane"
+	kubeadmutil "k8s.io/kubernetes/cmd/kubeadm/app/util"
+	configutil "k8s.io/kubernetes/cmd/kubeadm/app/util/config"
+	"k8s.io/kubernetes/pkg/api"
+)
+
+// NewCmdControlplane return main command for Controlplane phase
+func NewCmdControlplane() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "controlplane",
+		Short: "Generate all static pod manifest files necessary to establish the control plane and eventually a local etcd.",
```

Comment:
I started thinking about this, and think we should leave etcd out of the controlplane CLI at least for the moment.
Eventually we'd get etcd into its own phase and just make the "write static pod to disk" function to the util pkg.

phases/etcd/local.go (very simplified)
```go
func CreateEtcdStaticPodManifestFile() error {
    return kubeadmutil.WriteStaticPodToDisk(GetEtcdStaticPodObject())
}

func GetEtcdStaticPodObject() v1.Pod { }
```

phases/controlplane/controlplane.go
```go
func CreateControlPlaneStaticPodManifestFiles() error {
    // for each control plane pod, write to disk:
    kubeadmutil.WriteStaticPodToDisk(GetEtcdStaticPodObject())
}

func GetControlPlaneStaticPodObjects() []v1.Pod { }
```

util/staticpod.go
```go
func WriteStaticPodToDisk() error { }
```

Does this layout sound good to you?